---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# evalGSVAsig

<!-- badges: start -->
<!-- badges: end -->

It's common to have a signature of genes, then to give each sample in a bulk rna-seq dataset a score reflecting the activity of that signature, such as by GSVA.

The goal of evalGSVAsig is to identify which genes from a signature are contributing most to the sample signature activity scores across all samples.  

For example, one may have a signature of genes that are upregulated during the process of epithelial to mesencymal transition (EMT), which is associated with tumor metastasis.
One can use a tool (such as GSVA) to assign a score to each patient in the TCGA dataset based on each patient's gene expression profile, that corresponds to how 'active' the EMT signature is in each patient.  It would suggest that a patient with a higher score has more activity of the genes in the EMT geneset, and thus may have a more aggressive tumor.  Some genes will be very correlated with the score, suggesting their larger role in EMT in this particular set of samples, while some genes in the EMT signature may have very little correlation.

This package simply takes the correlation of each gene in the signature with the signature score, to identify the genes that are contributing most to the score.  

## Installation

You can install the development version of evalGSVAsig from [GitHub](https://github.com/) with:

``` r
# install.packages("devtools")
devtools::install_github("montilab/evalGSVAsig@dev")
```

## Example

This is a basic example which shows the format of the data:

```{r example, eval=FALSE, message=FALSE, warnings=FALSE }
library(evalGSVAsig)

## example gene list
sig <- c("Gene1", "Gene2", "Gene3")
signature_list <- c(list('signature1' = sig))

## run function. if sig_score = NULL, score will be provided. Otherwise provide pre-computed score with sig_score = score
output <- evalGSVAsig::omics_signature_projection(eset = eset, signature = signature_list)

## view output
## data.frame of genes ordered by correlation to signature score
print(output$score_cor)

## data.frame of samples ordered by signature score
print(output$sig_score)

#show heatmap of ALL gene expression in relation to GSVA score
print(output$heatmap_all_genes)

#show heatmap of SIGNATURE gene expression in relation to GSVA score
print(output$heatmap_sig_genes)
```

With sample data:

```{r plots, message=FALSE, warning=FALSE, results='hide' }
#with our sample data
data(signature)
data(eset)

output <- evalGSVAsig::omics_signature_projection(
  eset = eset, 
  signature = signature, 
  metacol = ComplexHeatmap::HeatmapAnnotation( 
    df = pData(eset) |> dplyr::select(Group),
    col = list(pheno = c(Treated = "pink", Control = "gray"))))
```

View tables:

```{r tables}
## Show some of the genes' correlation to the signature score
print(head(output$score_cor))
## Show some of the signature scores
print(head(output$sig_score))
```

View heatmap that plots all signature and non-signature genes:

```{r full.heatmap, dpi=300}
print(output$heatmap_all_genes)
```

View heatmap that plots only signature genes:

```{r sig.heatmap, dpi=300}
print(output$heatmap_sig_genes)
```

