---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)

pkgload::load_all(".", quiet = TRUE, export_all = FALSE, helpers = FALSE)
```

# sigProCon

<!-- badges: start -->
<!-- badges: end -->

It's common to have a signature of genes, then to give each sample in a bulk rna-seq dataset a score reflecting the activity of that signature, such as by GSVA.

The goal of sigProCon is to identify which genes from a signature are contributing most to the sample signature activity scores across all samples.  

For example, one may have a signature of genes that are upregulated during the process of epithelial to mesencymal transition (EMT), which is associated with tumor metastasis.
One can use a tool (such as GSVA) to assign a score to each patient in the TCGA dataset based on each patient's gene expression profile, that corresponds to how 'active' the EMT signature is in each patient.  It would suggest that a patient with a higher score has more activity of the genes in the EMT geneset, and thus may have a more aggressive tumor.  Some genes will be very correlated with the score, suggesting their larger role in EMT in this particular set of samples, while some genes in the EMT signature may have very little correlation.

This package simply takes the correlation of each gene in the signature with the signature score, to identify the genes that are contributing most to the score.  

## Installation

You can install the development version of sigProCon from [GitHub](https://github.com/) with:

``` r
# install.packages("devtools")
devtools::install_github("montilab/sigProCon@dev")
```

## Example with toy sample data:

```{r toy-example, eval=FALSE, message=FALSE, warning=FALSE}
data(toy_signature)
data(toy_eset)
print(dim(toy_signature))
print(dim(toy_eset))
```

### View toy data tables:

```{r toy-tables, message=FALSE, warning=FALSE}
toy_output <- sigProCon::signature_projection_contributors(
  eset = toy_eset,
  signature = list(signature1 = toy_signature)
)

## Show some of the genes' correlation to the signature score
print(head(toy_output$score_cor))

## Show some of the signature scores
print(head(toy_output$sig_score))
```

### View toy heatmap that plots all signature and non-signature genes:

```{r toy-full.heatmap, dpi=300}
print(toy_output$heatmap_all_genes)
```

### View toy heatmap that plots only signature genes:

```{r toy-sig.heatmap, dpi=300}
print(toy_output$heatmap_sig_genes)
```

### View toy k.s. plot of signature genes and scores:

```{r toy-ksplot, dpi=300}
print(toy_output$ks$plot)
```

## Example with package sample data (`eset` and `signature`):

```{r plots, message=FALSE, warning=FALSE, results='hide' }
#with our sample data
data(signature)
data(eset)

output <- sigProCon::signature_projection_contributors(
  eset = eset, 
  signature = signature)
```

### View tables:

```{r tables}
## Show some of the genes' correlation to the signature score
print(head(output$score_cor))
## Show some of the signature scores
print(head(output$sig_score))
```

### View heatmap that plots all signature and non-signature genes:

```{r full.heatmap, dpi=300}
print(output$heatmap_all_genes)
```

### View heatmap that plots only signature genes:

```{r sig.heatmap, dpi=300}
print(output$heatmap_sig_genes)
```

### View k.s. plot of signature genes and scores:

```{r ksplot, dpi=300}
print(output$ks$plot)
```
